src/Compilers/CSharp/Test/Syntax/Parsing/RefStructs.cs

```diff
@@ -94,27 +93,44 @@ class Program
             );
         }
 
+        [Fact]
+        public void DoublePartialModifier()
+        {
+            var test = @"
+partial partial ref struct Test {}
+";
+
+            var comp = CreateCompilation(test);
+            comp.VerifyDiagnostics(
+                // (2,9): error CS1525: Invalid expression term 'partial'
+                // partial partial ref struct Test {}
+                Diagnostic(ErrorCode.ERR_InvalidExprTerm, "partial").WithArguments("partial").WithLocation(2, 9),
+                // (2,9): error CS1002: ; expected
+                // partial partial ref struct Test {}
+                Diagnostic(ErrorCode.ERR_SemicolonExpected, "partial").WithLocation(2, 9),
+                // (2,1): error CS0246: The type or namespace name 'partial' could not be found (are you missing a using directive or an assembly reference?)
+                // partial partial ref struct Test {}
+                Diagnostic(ErrorCode.ERR_SingleTypeNameNotFound, "partial").WithArguments("partial").WithLocation(2, 1)
+                );
+        }
+
         [Fact]
         public void PartialRefStruct()
```

Comment:
Could you add a similar test with a difference in `ref` (one declaration of `S` with `ref` and the other without)?
May also be good to test in both orders (one without `ref` first vs. last).

Reply:
Let me know if the test generator coverage is sufficient.

Reply:
I should probably use InlineData for that.